[refactor] Define ThreadLocalVar

donaldong opened the pull request:
This is an attempt to dry up the thread-local vars. Currently, the key is defined manually and might have collisions.

katyho commented on lib/redis_memo/thread_local_var.rb:
```diff
@@ -0,0 +1,15 @@
+module RedisMemo::ThreadLocalVar
+  def self.define(var_name) # :nodoc:
+    thread_key = :"__redis_memo_#{var_name}__".freeze
+    const_set(var_name.to_s.upcase, thread_key)
```
Are these constants being used anywhere? Most call sites are using the method ThreadLocalVar.{var_name}?

donaldong replied:
Ah, I was using this to make sure var_names are uniquely defined globally.

If we define two ThreadLocalVars with the same var_name, there should be an error.

I will add a test case covering this behavior!

donaldong commented on lib/redis_memo.rb:
```diff
@@ -83,46 +84,46 @@ def self.deep_sort_hash(orig_hash)
+    # The connection attempts count and max connection attempts are reset in
+    # RedisMemo.with_max_connection_attempts
+    ThreadLocalVar.connection_attempts_count += 1
+    if ThreadLocalVar.connection_attempts_count > ThreadLocalVar.max_connection_attempts
```
🤦  huh I wonder how did this happen... totally unintentional. It's also interesting that the unit tests are still passing

donaldong commented:
add a test case, remove an unintentional update

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/redis-memo/compare/5b1f4ea7fb9cca0d65d5a15f26341adc6e7f51b4..72036beb7e887b07dfa1f36f80cfd99c943e21c7).
